apps/explorer/lib/explorer/migrator/heavy_db_index_operation/create_addresses_transactions_count_desc_partial_index.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Explorer.Migrator.HeavyDbIndexOperation.CreateAddressesTransactionsCountDescPartialIndex do
+  @moduledoc """
+  Create partial B-tree index on `addresses` table filtering by `fetched_coin_balance > 0`
+  and sorted by transactions_count DESC.
+  """
+
+  use Explorer.Migrator.HeavyDbIndexOperation
+
+  require Logger
+
+  alias Explorer.Chain.Cache.BackgroundMigrations
+  alias Explorer.Migrator.{HeavyDbIndexOperation, MigrationStatus}
+  alias Explorer.Migrator.HeavyDbIndexOperation.Helper, as: HeavyDbIndexOperationHelper
+
+  @table_name :addresses
+  @index_name "addresses_transactions_count_desc_partial"
+  @operation_type :create
+
+  @impl HeavyDbIndexOperation
+  def table_name, do: @table_name
+
+  @impl HeavyDbIndexOperation
+  def operation_type, do: @operation_type
+
+  @impl HeavyDbIndexOperation
+  def index_name, do: @index_name
+
+  @impl HeavyDbIndexOperation
+  def dependent_from_migrations, do: []
```

Comment:
For better safetiness I'd add dependency from DB operations in the same release:

```suggestion
  def dependent_from_migrations,
    do: [
      CreateAddressesVerifiedFetchedCoinBalanceDescHashIndex.migration_name()
    ]
```